packages/office-ui-fabric-react/src/components/Dialog/Dialog.tsx

```diff
@@ -79,21 +76,24 @@ export class Dialog extends BaseComponent<IDialogProps, IDialogState> {
         isClickableOutsideFocusTrap={ isClickableOutsideFocusTrap }
         isDarkOverlay={ isDarkOverlay }
         isOpen={ isOpen }
-        onDismiss={ onDismiss }
         onDismissed={ onDismissed }
         onLayerDidMount={ onLayerDidMount }
         responsiveMode={ responsiveMode }
+        { ...modalProps }
+        onDismiss={ onDismiss } // pull from contentProps.onDismiss once onDismiss is deprecated
```

Comment:
it's a callback that the content is able to call. I might have to leave it in and note that the prop is controlled by the component.